sky/utils/controller_utils.py

```diff
@@ -48,74 +49,95 @@
 class _ControllerSpec:
     """Spec for skypilot controllers."""
     name: str
-    cluster_name: str
+    # Use a list of strings to support fallback to old names.
+    candidate_cluster_names: List[str]
     in_progress_hint: str
     decline_cancel_hint: str
-    decline_down_when_failed_to_fetch_status_hint: str
+    _decline_down_when_failed_to_fetch_status_hint: str
     decline_down_for_dirty_controller_hint: str
-    check_cluster_name_hint: str
+    _check_cluster_name_hint: str
     default_hint_if_non_existent: str
     connection_error_hint: str
 
+    @property
+    def cluster_name(self) -> str:
+        """Returns the cluster name that exists."""
+        for candidate_name in self.candidate_cluster_names:
+            record = global_user_state.get_cluster_from_name(candidate_name)
+            if record is not None:
+                return candidate_name
```

Comment:
No, it is not possible, because:
1. if the old controller exist: we will always reuse the old one without creating a new job-controller
2. If the old controller does not exist: we will create a new one and always use the new job controller.